R/agglomerate.R

```diff
@@ -333,16 +333,14 @@ setMethod("agglomerateByVariable", signature = c(x = "SummarizedExperiment"),
 #' @export
 setMethod("agglomerateByVariable",
             signature = c(x = "TreeSummarizedExperiment"),
-            function(x, MARGIN, f, archetype = 1L, mergeTree = FALSE,
-                     mergeRefSeq = FALSE, ...){
+            function(x, MARGIN, f, archetype = 1L, mergeTree = FALSE, ...){
                 # Check MARGIN
                 MARGIN <- .check_MARGIN(MARGIN)
                 # Get function based on MARGIN
                 FUN <- switch(MARGIN, .merge_rows_TSE, .merge_cols_TSE)
                 # Agglomerate
                 x <- FUN(
-                    x, f, archetype = archetype, mergeTree = mergeTree,
-                    mergeRefSeq = mergeRefSeq, ...)
```

Comment:
But this disables the usage of mergeRefSeq; user cannot use that anymore, so this is wrong

Reply:
This is caused because this parameter is passed thorugh with "...".

The column-related functions do not have this parameter so this parameter flows with "...". Every time certain function is called, it check that the parameters match. Those paramaters that do not match, can flow with "...". If there is no "..." in function definition, the function throws error "unused argument"

Reply:
Sorry, my mistake. There is already mergeRefSeq in internal function, so this should actually work...

Reply:
ah yes. However,  mergeRefSeq is documented in agglomerat-methods and issues usage warning with the change I made. Unless it is ok to have mergeRefSeq undocumented. is it?

Reply:
You can put it under "..."

Reply:
In agglomerate-methods documentation

Reply:
Or one option is to add mergeRefSeq to agglomerateByRank. That function is only for rows so it would make sense. The function calls agglomerateByVariable internally

Reply:
okay. I add to agglomerateByRank then.